.infrastructure/helm-chart/todoapp/templates/app/deployment.yml

```diff
@@ -0,0 +1,92 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Chart.Name }}-deployment
+  namespace: {{ .Values.common.namespace }}
+spec:
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: {{ .Values.deployment.rollingUpdate.maxSurge }}
+      maxUnavailable: {{ .Values.deployment.rollingUpdate.maxUnavailable }}
+  selector:
+    matchLabels:
+      app: {{ .Values.todoapp.labelName }}
+  template:
+    metadata:
+      labels:
+        app: {{ .Values.todoapp.labelName }}
+    spec:
+      serviceAccountName: {{ .Values.serviceAccount.name }}
```

Comment:
Ensure that `.Values.serviceAccount.name` is defined in your `values.yaml` file to specify the service account for the deployment.